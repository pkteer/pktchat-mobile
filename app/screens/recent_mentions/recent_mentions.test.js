// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import RecentMentions from './recent_mentions';

describe('RecentMentions', () => {
    const baseProps = {
        actions: {
            clearSearch: jest.fn(),
            getPostThread: jest.fn(),
            getRecentMentions: jest.fn(),
            showPermalink: jest.fn(),
            selectPost: jest.fn(),
        },
        theme: Preferences.THEMES.denim,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <RecentMentions {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot when getRecentMentions failed', async () => {
        const error = new Error('foo');

        const newProps = {
            ...baseProps,
            actions: {
                ...baseProps.actions,
                getRecentMentions: jest.fn().mockResolvedValue({error}),
            },
        };
        const wrapper = shallowWithIntl(
            <RecentMentions {...newProps}/>,
        );

        await wrapper.instance().getRecentMentions();
        expect(wrapper.state('didFail')).toBe(true);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot when component waiting for response', () => {
        const error = new Error('foo');

        const newProps = {
            ...baseProps,
            actions: {
                ...baseProps.actions,
                getRecentMentions: jest.fn().mockResolvedValue({error}),
            },
        };
        const wrapper = shallowWithIntl(
            <RecentMentions {...newProps}/>,
        );

        wrapper.instance().getRecentMentions();
        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.state('isLoading')).toBe(true);
    });
});
