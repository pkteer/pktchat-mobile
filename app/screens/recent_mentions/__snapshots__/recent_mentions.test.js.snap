// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentMentions should match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
  testID="recent_mentions.screen"
>
  <Connect(StatusBar) />
  <Connect(ChannelLoader)
    channelIsLoading={true}
  />
</RNCSafeAreaView>
`;

exports[`RecentMentions should match snapshot when component waiting for response 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
  testID="recent_mentions.screen"
>
  <Connect(StatusBar) />
  <Connect(ChannelLoader)
    channelIsLoading={true}
  />
</RNCSafeAreaView>
`;

exports[`RecentMentions should match snapshot when getRecentMentions failed 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
  testID="recent_mentions.screen"
>
  <Connect(StatusBar) />
  <Connect(FailedNetworkAction)
    onRetry={[Function]}
    theme={
      Object {
        "awayIndicator": "#ffbc1f",
        "buttonBg": "#1c58d9",
        "buttonColor": "#ffffff",
        "centerChannelBg": "#ffffff",
        "centerChannelColor": "#3f4350",
        "codeTheme": "github",
        "dndIndicator": "#d24b4e",
        "errorTextColor": "#d24b4e",
        "linkColor": "#386fe5",
        "mentionBg": "#ffffff",
        "mentionColor": "#1e325c",
        "mentionHighlightBg": "#ffd470",
        "mentionHighlightLink": "#1b1d22",
        "newMessageSeparator": "#cc8f00",
        "onlineIndicator": "#3db887",
        "sidebarBg": "#1e325c",
        "sidebarHeaderBg": "#192a4d",
        "sidebarHeaderTextColor": "#ffffff",
        "sidebarTeamBarBg": "#14213e",
        "sidebarText": "#ffffff",
        "sidebarTextActiveBorder": "#5d89ea",
        "sidebarTextActiveColor": "#ffffff",
        "sidebarTextHoverBg": "#28427b",
        "sidebarUnreadText": "#ffffff",
        "type": "Denim",
      }
    }
  />
</RNCSafeAreaView>
`;
